ethereum/core/src/main/java/org/hyperledger/besu/ethereum/mainnet/AbstractBlockProcessor.java

```diff
@@ -67,26 +68,49 @@ TransactionReceipt create(
     private final boolean successful;
 
     private final List<TransactionReceipt> receipts;
+    private final Optional<List<TransactionReceipt>> privateReceipts;
```

Comment:
NIT: Do we really need to have this list as an optional? Can't we only check if it is empty instead?

Reply:
Done